### ISSUE TITLE
Error when using a non-start disk for location of personal files.

### ISSUE THREAD

**Issue by EinarMostad** [lxde/lxqt#1389]
I have two disks in my laptop, a small SSD for the operating system and programs and a large HDD for my files. Both have been added to /etc/fstab to show up on boot. When I tried to add my actual pictures, documents... folders from my HDD, I got an error. I don't know why, maybe it has to do with the names of these folders? I use spaces, numbers and non-ASCII characters in the folder names. Could that be the source of the problem?

Just to make it clear: From the picture, it looks like I am using folders in my /home/einar/ folder, but I am not. The names of the folders changed back to the defaults as the error came up. The folders I tried to set was on another disk mounted at boot to /run/media/einar/Magnus_Super/ and they open up fine in PCManFM. I am not using the HDD for my ~, but I am using folders on it to store my files.

![error when using a non-startdisk for locations for personal files](https://user-images.githubusercontent.com/33556631/32675908-df77ce2c-c658-11e7-9115-6d007d8ba560.png)

##### Expected Behavior
I should be able to set these folders to any folder mounted on my system.

##### Current Behavior
I get the error only when trying to set folders on my HDD as the personal files folder.

##### Steps to Reproduce (for bugs)
Open LXQt session settings, change folders to folders on another disk.

##### System Information

* Distribution & Version: Arch Linux
* Kernel: 
* Qt Version: 
* liblxqt Version:  0.12.0-1
* Package version:   0.12.0-1

**Comment by agaida** [comment]
there is a similar bug in Launchpad

**Comment by tsimonq2** [comment]
For the record, [this](https://bugs.launchpad.net/ubuntu/+source/pcmanfm-qt/+bug/1768961) is the LP bug report @agaida is referencing.

**Comment by scott092707** [comment]
The bug report referenced by tsimonq2 is mine, and the bug still exists post-release of Lubuntu 18.10.

I also attempted to change my user directories with the session-settings->user directories app, and
had the same error message upon "Close" that is shown in the above picture.

[Since this did not work, I went back to my usual manual change of user-dirs.dirs, followed by log out/in.
I note that the change in user-dirs.dirs did NOT alter what was shown in session-settings->user directories.  It still showed "/home/scott/..." ]

I also note that each time the browse icon is clicked for a directory, it forces one to start at /home/<user> .
Since probably all user directories are in the same containing directory, if the user navigates to a different containing directory, the next time he clicks on "browse", it might for the user's convenience, default to the same containing directory the user browsed to before.
Since there are eight user directories to change, this means for me, that I need to click:
 up / up / up to get to "/", then double click on /data, and then double click on /data/scott,
and finally double click on the relevant user directory (or select the directory and click on "Choose")
eight different times to change all the user directories.
Alternatively, you could have a radio button that if clicked, would change all containing directories to the same one.  (If subsequently un-checked, one could change one or two oddball directories that didn't fit the pattern of all user directories being in the same containing directory.)
Or something like that...

**Comment by agaida** [comment]
Ok, we should close this now - @tsujan, do you agree? - Maybe there should be a follow up bug to make the messages more verbose. 

IMHO Folders outside the $HOME should not be considered as valid settings for xdg user dirs. - Maybe i'm wrong, one should check the spec about.

**Comment by agaida** [comment]
We should improve the message that way that directories outside home are not allowed. things like symbolic links or (better) bind mounts wasn't invented yesterday.

**Comment by tsujan** [comment]
> We should improve the message 

Yes, "An error occurred..." is too general when we know what has happened in the code.

BTW, I'm not so brave to loosen that condition.

**Comment by tsujan** [comment]
Oh, my bad: we don't know what has happened because there are 3 cases of return false;

Of course, the code can be changed so that messages are more specific but I'm not sure it's worth the effort for such rare cases.

NOTE: GitHub got confused AGAIN! This is alarming.

**Comment by tsimonq2** [comment]
> We require that dirs should be inside home: https://github.com/lxqt/libqtxdg/blob/7d7e6e7a898529feb1293ceb9c4a3c6be083fac4/src/qtxdg/xdgdirs.cpp#L182 . So, the error message is expected. Whether this condition can be loosened is another question but one can guess that it may not be safe.

But why? I can't think of a good reason why this is the way it is.

**Comment by tsujan** [comment]
> I can't think of a good reason

Because you're very brave.

**Comment by tsujan** [comment]
> very brave != dont care

Right.

I'm not sure about Qt's reaction. It should be thoroughly tested by someone (not me; I can't).

**Comment by tsujan** [comment]
Qt aside (it shouldn't complain, I think), older codes may suppose that those folders are inside `$HOME`. I'm not even sure that we don't have such codes.

**Comment by agaida** [comment]
@tsimonq2 made yesterday a noteworthy comment about upgrading from LXDE to LXQt - "We don't support it." - So i would like to make the same comment here and now: We just don't support that, there are alternatives. Educate the users about and be done with.

**Comment by tsimonq2** [comment]
> @tsimonq2 made yesterday a noteworthy comment about upgrading from LXDE to LXQt - "We don't support it."

If you're going to quote me, quote me correctly. :stuck_out_tongue:

I said that installing LXQt on an 18.04 system and LXDE on 18.10 isn't supported; I said nothing about 18.04 -> 18.10 upgrades, that's what our documentation is for.

I'll do some testing in Lubuntu with removing that conditional. If it breaks nothing (which it won't :stuck_out_tongue:) I'll report back.

**Comment by tsujan** [comment]
> If it breaks nothing...

So, you know that you need to test it as your system and with those folders outside your home and with as many apps as you could install for at least a month.

**Comment by tsujan** [comment]
> We just don't support that, there are alternatives.

I agree. @tsimonq2 thinks just "doing some tests" is enough.

**Comment by scott092707** [comment]
I guess people weren't understanding the rationale of what I was doing...

When I set up my computer (lo, those many years ago...), I read something about how to set up my partitioning.
I have 5 different partitions in which I can install OSes (originally, to try different distros, but long since merely for subsequent versions of the same distro - Lubuntu).
If an install goes kerflewy, I can just boot into the previous one and try and find out what is wrong and try and fix it.  Also, I can just re-purpose an earlier distro's partition to test an OS's development version,
and still use my usual OS partition for day-to-day work.

What I read was, that even if I keep my personal data in a separate partition, it is still a good idea to have $HOME in the OS's install partition, because any installed app will put its config folders in $HOME, and from one distro version to the next, the config data may change.  If $HOME resides with my personal data in a separate partition, then each install will wipe out and replace the app config data, so that if I try to go back and use a previous distro version, the apps may not work correctly, as they are now trying to run with later config data. (And unless I rename all my personal folders to something other than the standard names, I suppose they would get erased, too...)

So, saying that my personal data MUST be in $HOME does not make logical sense (to me, at least...):
that would mean that either I make my install partition the whole drive (so there's room for all my personal data), and therefore have no way to boot multiple OS's, or $HOME is in my data partition, and the app config files get munged every time I test a new distro (or fail to install a new version, and try to go back to a previous one).
Neither way is very appetizing...

**Comment by agaida** [comment]
@scott092707 - you might be surprised, i understand the rationale very well, i use this setup with minor variations since eight years and wrote endless rants about. To make it short: It seems to me that you missed some points about *nix filesystems and the used tree structure. Nothing important, but worth to think about it. Otherwise: Congratulations for a fine setup, right now my laptop and my workstation with some installations are built that way (approx 12 different installations, arch, debian, siduction, lubuntu, freebsd included)

Right now i have a small 500G SSD called `datagrave` in the workstation among other drives: 
```
mount /dev/sd?1 /mnt/datagrave
----
mount --bind /mnt/datagrave/agaida/work /home/agaida/work
mount --bind/mnt/datagrave/agaida/Documents /home/agaida/Documnets
# and so on
```
Voila, my files are in my $HOME, done - as i wrote above, linking would work too.

**Comment by scott092707** [comment]
@agaida 
Thank you for your comments on my setup.
I am now using Debian testing, but still using LXQt.

I will probably soon try using a bind mount for my personal folders.

I don't quite know the difference between bind and rbind, for my fstab lines.
I THINK I don't need rbind unless I have a mount that is contained within another mount, but...

"using user mounts (far better)"  I have not yet seen that term - what is it, and how does it differ from bind/rbind?


I wonder also, if a linux distro could just automatically look at user-dirs.dirs, and for each folder that did not match the default entry, create a fstab (mtab?) mount binding it to the default entry...

What was the original purpose of user-dirs, unless it was to allow the user to say where he wanted the dirs to be; if we are now saying - "No, you cannot do that.", then why?
